pkg/azuredisk/azuredisk_v2.go

```diff
@@ -62,6 +62,7 @@ var controllerLeaseDurationInSec = flag.Int("lease-duration-in-sec", 15, "The du
 var controllerLeaseRenewDeadlineInSec = flag.Int("lease-renew-deadline-in-sec", 10, "The duration that the acting controlplane will retry refreshing leadership before giving up.")
 var controllerLeaseRetryPeriodInSec = flag.Int("lease-retry-period-in-sec", 2, "The duration the LeaderElector clients should wait between tries of actions.")
 var partitionLabel = "azdrivernodes.disk.csi.azure.com/partition"
+var isTestRun = flag.Bool("is-test-run", false, "Boolean flag to indicate whether this instance is being used for sanity or integration tests")
```

Comment:
As part of sanity and integration tests, we are creating a vm on azure and passing it's nodeId in the csi driver so that the attachments and mount can happen on that node. (We have been doing it this way because we can't attach disks to non azure nodes.)

Since that vm doesn't belong to a K8s cluster, the check on line 338(which takes the nodeId of the azure vm) fails and the AzDriverNode instance cannot be created. I've added this flag so that we can still have an AzDriverNode CRI. The AzDriverNode instance hence created will represent the minikube node.

Reply:
Ah, yes, of course. Make sense.